src/Normalizer/EntityReferenceItemNormalizer.php

```diff
@@ -98,6 +103,16 @@ public function normalize($field_item, $format = NULL, array $context = []) {
         $arguments = isset($field_mappings['datatype_callback']['arguments']) ? $field_mappings['datatype_callback']['arguments'] : NULL;
         $values_clean['@value'] = call_user_func($callback, $target_entity, $arguments);
       }
+      $field = $field_item->getParent();
+      $field_context = $this->jsonldContextgenerator->getFieldsRdf(
+        $context['current_entity_rdf_mapping'],
+        $field->getName(),
+        $field->getFieldDefinition(),
+        $context['namespaces']
+      );
+      if (isset($field_context[$field_keys[0]])) {
+        $values_clean = $values_clean + $field_context[$field_keys[0]];
```

Comment:
I would guess that initializing  $values_clean to an empty array could work here. or moving  $values_clean['@id'] = key($embedded['@graph']); before all conditions are applied (just a few lines up)